aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -45,6 +45,22 @@ static void sigmoid_kernel(TensorIterator& iter) {
   });
 }
 
+static void fill_kernel(TensorIterator& iter, Scalar value) {
+  if (iter.dtype() == at::ScalarType::Half) {
+    at::Half fill_value = value.to<at::Half>();
+    unary_kernel(iter, [=](at::Half a) -> at::Half { return fill_value; });
+  } else {
+    AT_DISPATCH_ALL_TYPES(iter.dtype(), "fill_cpu", [&]() {
+      scalar_t fill_value = value.to<scalar_t>();
+      Vec256<scalar_t> fill_vec_value = Vec256<scalar_t>(fill_value);
+      unary_kernel_vec(
```

Comment:
This shouldn't be `unary_kernel`. `fill_` has zero tensor inputs. That's what's causing the undefined behavior. It's only getting reported with bool tensor, but I think it's undefined behavior in general. You'll need a "nullary" kernel in Loops.h. (There's already a CUDA version, but no CPU version).

You should be able to copy-paste unary_kernel and just start deleting code.

It would also be good to add an `AT_ASSERT(iter.ntensor() > 1)` in unary_kernel/unary_kernel_vec.

Reply:
Since this is blocking for some downstream work I'll mark that for an upcoming PR. 

cc @VitalyFedyunin for visibility. Let's add this as a work item.

Reply:
This cannot be landed as is. `unary_kernel` and `unary_kernel_vec` are accessing memory passed the end of the array.

Reply:
Copy have no problems about having `AT_ASSERT(iter.ntensors() > 1);` inside of unary_kernel and unary_kernel_vec.